Library/OcBootManagementLib/BootEntryInfo.c

```diff
@@ -478,11 +482,15 @@ OcGetBootEntryIcon (
   *ImageData = NULL;
   *DataLength = 0;
 
-  if (BootEntry->Type == OC_BOOT_EXTERNAL_TOOL || BootEntry->Type == OC_BOOT_RESET_NVRAM) {
+  if ((BootEntry->Type & OC_BOOT_SYSTEM) != 0) {
```

Comment:
Is this legit? Sounds like it must be an ASSERT again.

Reply:
I don't think so? This says 'if the boot entry is a system tool' (OC_BOOT_SYSTEM is an OR of all OC_BOOT_ bit values which are system tools) - should be okay? (Also, is not a direct replacement for what it replaces, that is split into two, above and just below.)

```
#define OC_BOOT_RESET_NVRAM         BIT8
#define OC_BOOT_TOGGLE_SIP          BIT9
#define OC_BOOT_SYSTEM              (OC_BOOT_RESET_NVRAM | OC_BOOT_TOGGLE_SIP)
```

Reply:
(The assert is when the internal tool _name_ is invalid for the tool type.)

Reply:
I mean, can this function be called for such entries?

Reply:
Yes, it always was. And it in fact returned values for them - the red stuff that I removed in `OcToolDescribeEntry`. This _is_ `CustomDescribe`. It's arguable whether it needs to be called for them now, but certainly on the current path this is legal. And made more sense when there was more in  `OcToolDescribeEntry` - but all recent changes are making more and more code drop out of that - possibly to the extent of it disappearing completely.

Reply:
At which point you get a line like the above, saying 'nothing to do here'. But not illegal call.